### DISCUSSION TITLE
What is the best format and bpp for performance in headless client?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8211] [Q&A] [answered]
hello,

I am using a headless sample client to produce the desired function.

I copy and use the required image from `end_paint` using `freerdp_image_copy`.

That's why I was able to get the data in the format I wanted, regardless of the format format specified in `gdi_init`.

And I checked some articles that lower bpp helps performance when looking for information related to freerdp's performance.

When using `freerdp_image_copy`, the format I need is `RGB24`.

How should I set the format(`gdi_init`) and bpp to get the best performance?

Do I have to set it the same as the format I want to copy? Or is it better to set it lower? Or doesn't it matter?

**Comment by akallabeth** [comment, marked as answer]
* Performance differences in RDP sessions are negligible, so best always use `32bpp` (except you really have some very old system like `windows XP`) most modern sessions use `rfx` (windows 7 and newer) or `gfx` (windows 8 and newer) channel which is 32bpp anyway. (at least if you have implemented your client correctly)
* Choose a format for `gdi_init` which is compatible with the one you need later on (conversion is expensive, so here you can actually speed things up)
* with `RGB24` you mean real 3 byte per pixel bitmap or a `RGBX` 4 byte per pixel bitmap with alpha channel at don´t care?

**Reply by kimcoldwater** [reply]
Thank you for your quick reply.

Yes, what I need is `3 bytes per pixel bitmap`.

I think I will try to connect to rdp on Windows 8 or higher.

Should I apply rfx or gfx separately at this time?

i not use sound and media streaming.

I know these are hardware-related options, and I looked for relevant information, 

but I still didn't really understand what freerdp does.

Can you briefly explain the function? 

And Is it good to turn on the rfx option?

**Reply by akallabeth** [reply]
RFX/GFX are connection options, your client just needs to announce support and you will get the image updates in `EndPaint`. 

Server will just use different ways to transfer data (with `gfx` eigher as `ClearCodec`, `RFX`, `Progressive` or `H264` data).
that will reduce bandwidth requirements drastically (specially if you use higher resolutions)

As for `what freerdp does, it does all the connection management and data encoding/decoding of RDP ;)
